### PR TITLE
LPS-89210 anonymization deletes versions then persists them as new, deleteLayout method now use the layoutLocalServiceBase delete method

### PR DESCRIPTION
**Pull request by zoltancsaszi**

### CODE REVIEW

**Comment by zoltancsaszi** [comment]
ci:test:sf

**Comment by zoltancsaszi** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java [hunk 1]
```diff
@@ -608,8 +608,12 @@ public void anonymizeLayout(Layout layout, long userId, User anonymousUser)
 		layoutVersionStream.forEach(
 			layoutVersion -> {
 				if (layoutVersion.getUserId() == userId) {
+					layoutVersionPersistence.removeByPlid(
+						layoutVersion.getPlid());
```
Comment (Preston-Crary) [inline portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java]: You're deleting things you should not delete that are not being added back.  Please be very careful with deletion logic, there's no recovery for lost data besides a DB backup which if the issue isn't caught quickly enough could be really bad.

There's no reason to delete by anything you have what you want to delete.

Comment (zoltancsaszi) [reply portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java]: Oh, sorry. My mistake. My original intention was to use normal the remove(...) method.

https://github.com/Preston-Crary/liferay-portal/pull/689

**Pull request closed without merging**